daemon/algod/api/server/v2/handlers.go

```diff
@@ -763,6 +763,23 @@ func (v2 *Handlers) GetStatus(ctx echo.Context) error {
 		return internalError(ctx, err, errFailedRetrievingNodeStatus, v2.Log)
 	}
 
+	ledger := v2.Node.LedgerForAPI()
+	latestBlkHdr, err := ledger.BlockHdr(ledger.Latest())
+	if err != nil {
+		return internalError(ctx, err, errFailedRetrievingLatestBlockHeaderStatus, v2.Log)
+	}
+
+	votesToGo := uint64(latestBlkHdr.NextProtocolVoteBefore) - uint64(stat.LastRound)
+	consensus := config.Consensus[protocol.ConsensusCurrentVersion]
+	upgradeVoteRounds := consensus.UpgradeVoteRounds
+	upgradeThreshold := consensus.UpgradeThreshold
+	votes := uint64(upgradeVoteRounds) - votesToGo
+	votesYes := latestBlkHdr.UpgradeState.NextProtocolApprovals
+	votesNo := votes - votesYes
+
+	upgradePropose := string(latestBlkHdr.UpgradeVote.UpgradePropose)
+	upgradeDelay := uint64(latestBlkHdr.UpgradeVote.UpgradeDelay)
```

Comment:
I think most of this should move into the `upgradePropose != ""` section below. Alternatively, it might make sense to implement it in `Node.Status()`.